### PR TITLE
Bump dependencies

### PR DESCRIPTION
**Pull request by pellared**
I plan to create a release after this PR is merged.

### CODE REVIEW

File: distro/otel.go [hunk 1]
```diff
@@ -26,7 +26,7 @@ import (
 	"go.opentelemetry.io/otel/sdk/metric"
 	"go.opentelemetry.io/otel/sdk/resource"
 	"go.opentelemetry.io/otel/sdk/trace"
-	semconv "go.opentelemetry.io/otel/semconv/v1.17.0"
+	semconv "go.opentelemetry.io/otel/semconv/v1.21.0"
```
Comment (pellared) [inline distro/otel.go:29]: I decided to bump only semconv for resource like in upstream.

File: distro/otel_test.go [hunk 2]
```diff
@@ -48,7 +48,7 @@ import (
 
 const (
 	spanName   = "test span"
-	metricName = "test instrument"
+	metricName = "test_instrument"
```
Comment (pellared) [inline distro/otel_test.go:51]: Whiteout it the tests were failing with `invalid instrument name: test instrument: must only contain [A-Za-z0-9_.-]`

File: distro/otel.go (hunk 1 above)
Comment (MrAlias) [reply distro/otel.go:29]: Can you open an issue to track upgrading the rest of the repository?

File: distro/exporter.go [hunk 3]
```diff
@@ -21,7 +21,7 @@ import (
 	"os"
 
 	"github.com/go-logr/logr"
-	"go.opentelemetry.io/otel/exporters/jaeger"
+	"go.opentelemetry.io/otel/exporters/jaeger" //nolint:staticcheck // Jaeger is deprecated, but we still support it to not break existing users.
```
Comment (MrAlias) [inline distro/exporter.go:24]: Can you open an issue to track our deprecation of this? IIRC it was deprecated in the GDI spec.

**Review by MrAlias — APPROVED** [review]

File: distro/exporter.go (hunk 3 above)
Comment (pellared) [reply distro/exporter.go:24]: I will create a deprecation log message as part of resolving https://github.com/signalfx/splunk-otel-go/issues/2484

File: distro/otel.go (hunk 1 above)
Comment (pellared) [reply distro/otel.go:29]: https://github.com/signalfx/splunk-otel-go/issues/2486

**Pull request merged**